src/swaps/EvmSwapProvider.ts

```diff
@@ -38,12 +38,12 @@ export abstract class EvmSwapProvider extends SwapProvider {
     const client = this.getClient(network, walletId, quote.from, quote.fromAccountId) as Client<EvmChainProvider>;
     const signer = client.wallet.getSigner();
 
-    const tokenContract = Typechain.ERC20__factory.connect(fromAsset.contractAddress, signer);
+    const tokenContract = Typechain.ERC20__factory.connect(fromAsset.contractAddress.toLowerCase(), signer);
```

Comment:
No need for that because `fromAsset` comes from the `assetsAdapter` where the lowerCase is performed.